src/libsystemd/sd-network/sd-network.c

```diff
@@ -171,6 +171,18 @@ _public_ int sd_network_link_get_operational_state(int ifindex, char **state) {
         return network_link_get_string(ifindex, "OPER_STATE", state);
 }
 
+_public_ int sd_network_link_get_required_for_online(int ifindex) {
+        char* s;
+        int ret;
+
+        ret = network_link_get_string(ifindex, "REQUIRED_FOR_ONLINE", &s);
+
+        if (ret < 0)
+             return ret;
+
+        return strcmp(s, "true") == 0;
```

Comment:
Just use `return parse_boolean(s)`. This makes things more resilient if the format ever changes.

Reply:
Still needs r < 0, no? Otherwise parse_boolean might assert if for some reason network_link_get_string() returns -EINVAL?